lib/ansible/modules/identity/ipa/ipa_role.py

```diff
@@ -31,38 +31,53 @@
     - If an empty list is passed all assigned hosts will be unassigned from the role.
     - If option is omitted hosts will not be checked or changed.
-    - If option is passed all assigned hosts that are not passed will be unassigned from the role.
+    - Any assigned hosts which are not passed will be unassigned from the role.
+    - If the state is set to 'add_members', any provided hosts not currently in the role will be added.
+    - If the state is set to 'remove_members', any provided hosts currently in the role will be removed.
   hostgroup:
     description:
     - List of host group names to assign.
+    - If the state is set 'present', the provided host groups will be assigned to the role.
     - If an empty list is passed all assigned host groups will be removed from the role.
     - If option is omitted host groups will not be checked or changed.
-    - If option is passed all assigned hostgroups that are not passed will be unassigned from the role.
+    - Any assigned host groups which are not passed will be unassigned from the role.
+    - If the state is set to 'add_members', any provided host groups not currently in the role will be added.
+    - If the state is set to 'remove_members', any provided host groups currently in the role will be removed.
   privilege:
     description:
     - List of privileges granted to the role.
+    - If the state is set 'present', the provided privileges will be assigned to the role.
     - If an empty list is passed all assigned privileges will be removed.
     - If option is omitted privileges will not be checked or changed.
-    - If option is passed all assigned privileges that are not passed will be removed.
+    - Any assigned privileges which are not passed will be unassigned from the role.
+    - If the state is set to 'add_members', any provided privileges not currently in the role will be added.
+    - If the state is set to 'remove_members', any provided privileges currently in the role will be removed.
     version_added: "2.4"
   service:
     description:
     - List of service names to assign.
+    - If the state is set 'present', the provided services will be assigned to the role.
     - If an empty list is passed all assigned services will be removed from the role.
     - If option is omitted services will not be checked or changed.
-    - If option is passed all assigned services that are not passed will be removed from the role.
+    - Any assigned services which are not passed will be unassigned from the role.
+    - If the state is set to 'add_members', any provided services not currently in the role will be added.
+    - If the state is set to 'remove_members', any provided services currently in the role will be removed.
   state:
     description: State to ensure
     default: "present"
-    choices: ["present", "absent"]
+    choices: ["present", "absent", "add_members", "remove_members"]
```

Comment:
I wouldn't overload `state`, but add a new option, perhaps called `exclusive` (feel free to pick a better name)

Reply:
What options would you expect to use to remove members from a role?  The 'absent' state in conjunction with the new option?